src/modules/aws_ec2/mappers/instance_block_device_mapping.ts

```diff
@@ -0,0 +1,259 @@
+            const instances = await this.module.instance.db.read(ctx);
+            for (const instance of instances) {
+              if (!instance.instanceId) continue; // if we do not have an id we skip
+
+              // check instance block device mappings
+              try {
+                const mapping = await this.module.instance.getInstanceBlockDeviceMapping(
+                  client.ec2client,
+                  instance.instanceId,
+                );
+
+                for (const newMap of mapping ?? []) {
+                  if (newMap.DeviceName && newMap.Ebs?.VolumeId) {
+                    const volume = await this.module.generalPurposeVolume.db.read(
+                      ctx,
+                      this.module.generalPurposeVolume.generateId({
+                        volumeId: newMap.Ebs.VolumeId ?? '',
+                        region,
+                      }),
+                    );
+                    const res = await this.instanceBlockDeviceMappingMapper(newMap, instance, ctx);
+                    if (res) out.push(res);
+                  }
+                }
+              } catch (e: any) {
+                // it can be the case that it exists on the db but is not found on the cloud
+                if (e.Code === 'InvalidInstanceID.NotFound') continue;
+                else throw new Error(e);
+              }
+            }
+          }),
+        );
+        return out;
+      }
+    },
+    update: async (es: InstanceBlockDeviceMapping[], ctx: Context) => {
+      const out = [];
+      for (const e of es) {
+        const client = (await ctx.getAwsClient(e.region)) as AWS;
+        if (!e.volume || !e.volume.volumeId) continue; // cannot update a mapping without a volume
+        const cloudRecord = ctx?.memo?.cloud?.InstanceBlockDeviceMapping?.[this.entityId(e)];
+
+        if (e.deleteOnTermination !== cloudRecord.deleteOnTermination) {
+          // we need to update the delete on termination for that specific device
+          const mappings = await this.module.instance.getInstanceBlockDeviceMapping(
+            client.ec2client,
+            e.instance.instanceId,
+          );
+          for (const newMap of mappings ?? []) {
+            if (newMap.DeviceName === e.deviceName && newMap.Ebs?.VolumeId === e.volume?.volumeId)
+              newMap.Ebs!.DeleteOnTermination = e.deleteOnTermination;
+            const result = await this.modifyBlockDeviceMapping(
+              client.ec2client,
+              e.instance.instanceId,
+              mappings,
+            );
+            if (result) out.push(e);
+          }
+        } else {
+          // the only case for update is to change the device name, we will need to delete and recreate
+          await this.module.instanceBlockDeviceMapping.cloud.delete(e, ctx);
+          const newMap = await this.module.instanceBlockDeviceMapping.cloud.create(e, ctx);
+          if (!newMap) continue;
```

Comment:
since we have this case here we should also have the informative fn `updateOrReplace` to show to the users correctly when it is an update and when a replace